### ISSUE TITLE
Dev mode fails if jvmArgs contain unresolvable placeholder

### ISSUE THREAD

**Issue by famod** [quarkusio/quarkus#8837]
**Describe the bug**
Given following configuration for `quarkus-maven-plugin`:
```xml
          <configuration>
            <jvmArgs>-Dfoo=${missing}</jvmArgs>
          </configuration>
```
makes the dev mode fail with:
- Quarkus 1.3.1 & 1.3.2:
    ```
    [INFO] --- quarkus-maven-plugin:1.3.2.Final:dev (default-cli) @ modmono-quarkus-dist ---
    [INFO] Changes detected - recompiling the module!
    [INFO] Compiling 1 source file to C:\Develop\_dev\modmono-quarkus-github\dist\target\classes
    Listening for transport dt_socket at address: 5005
    2020-04-24 22:47:51,061 INFO  [org.jbo.threads] (main) JBoss Threads version 3.0.1.Final
    2020-04-24 22:47:51,160 ERROR [io.qua.dev.DevModeMain] (main) Failed to start Quarkus: java.lang.IllegalArgumentException: Undeclared build item class io.quarkus.deployment.builditem.ApplicationClassNameBuildItem
            at io.quarkus.builder.Messages.undeclaredItem(Messages.java:13)
            at io.quarkus.builder.BuildResult.consume(BuildResult.java:48)
            at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:79)
            at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:60)
            at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:236)
            at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:39)
            at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:131)
            at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
            at io.quarkus.dev.DevModeMain.start(DevModeMain.java:113)
            at io.quarkus.dev.DevModeMain.main(DevModeMain.java:54)
    
    2020-04-24 22:47:51,160 INFO  [io.qua.dev.DevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
    ```
- Quarkus 1.4.1 (haven't checked 1.4.0):
    ```
    2020-04-24 20:15:07,628 ERROR [io.qua.dep.dev.DevModeMain] (main) Failed to start Quarkus: java.lang.RuntimeException: java.lang.IllegalStateException: No root definition found for class io.quarkus.runtime.BannerRuntimeConfig
            at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:247)
            at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:130)
            at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:52)
            at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:61)
            at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:265)
            at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:40)
            at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:129)
            at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:82)
            at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:116)
            at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:56)
    Caused by: java.lang.IllegalStateException: No root definition found for class io.quarkus.runtime.BannerRuntimeConfig
            at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadResult.requireRootDefinitionForClass(BuildTimeConfigurationReader.java:795)
            at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:219)
            at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:98)
            at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:245)
            ... 9 more
    ```
    Instead of `BannerRuntimeConfig` I have also seen `ThreadPoolConfig` and others.

**Expected behavior**
Dev mode should not fail. Maybe it could print a warning?

**Actual behavior**
Dev mode fails with strange errors that leave the user clueless.

**To Reproduce**
Unfortunately, the following steps can only reproduce the problem for 1.3.2. I was unable to provoke the 1.4.1 error with it, but it fails reproducibly in my current main project (closed source).
1. Clone https://github.com/famod/modmono-quarkus.git
2. Switch to branch `quarkus-8837-missing-jvmarg`
3. Build and start dev mode:
    ```
    mvn clean install -DskipTests && mvn quarkus:dev -f dist
    ```

**Configuration**
n/a

**Screenshots**
n/a

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:
    ```MINGW64_NT-10.0-18363 W4DEUMSY9003463 3.0.7-338.x86_64 2019-11-21 23:07 UTC x86_64 Msys```
 - Output of `java -version`:
    ```
    openjdk version "11.0.6" 2020-01-14
    OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
    OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
    ```
 - GraalVM version (if different from Java): n/a
 - Quarkus version or git rev: `1.3.2.Final` or `1.4.1.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): `Apache Maven 3.6.3`

**Additional context**
n/a

**Comment by famod** [comment]
Still happens in 1.4.2.Final in a project that I am now allowed to share.

**Comment by jaikiran** [comment]
I can't reproduce this.  Looks like you are on Windows and perhaps this is Windows OS specific? I don't have any Windows system at hand right now.

**Comment by famod** [comment]
For some strange reason I cannot reproduce it myself (on Windows) with the reproducer that I linked above.
But I can reproduce it in a more complex, closed source project.

**Comment by famod** [comment]
Does not happen anymore with 1.6.0.CR1. Might have been fixed in 1.5.x already (didn't check).
